adrs/0541531435/README.md

```diff
@@ -0,0 +1,118 @@
+# NodeJS source code directory structure
+
+* **State:** Draft
+* **Created:** 2020-11-21
+* **Tags:** nodejs,directory
+
+## Context
+
+There are no guidelines on where to put the source code for your NodeJS
+projects.
+
+* Do I suppose to use `src`, `lib`, `dist` or `build` directories, or maybe put
+everything at the root of the project?
+
+Not having a clear guideline adds cognitive load, as well as add friction
+contributing to projects, and sometimes relying on a particular group of people
+that know the unspoken/undocumented rule.
+
+In other completely valid cases, like putting everything at the root of the
+project, don't take into consideration caveats like having so many metadata such
+as `.gitignore`, `.prettierrc`, `.eslintrc`, `.markdownlint.json`,
+`tsconfig.json` files and many other files and directories.
+
+Subjectively speaking, they add noise and hard to navigate or concentrate when
+put all your files at the root of the project.
```

Comment:
I think we can describe the purpose of the loose files in the root of a directory.  I'll take a stab: these files are configuration for 3rd party tools, including git, GitHub, package managers, ci, linters, build tools, dev tools, and others.

Separating our first-party code into its own directory brings clarity to the intention of both the root directory and the src directory.

Reply:
Out-of-scope, some of those files location are enforced by the tooling behind, so we cant do anything sometimes.